### PR TITLE
Add test cases to coverage all events cases

### PR DESCRIPTION
**Pull request by stockiNail**
This PR adds test cases on events for all annotation types.

The display option test spec has been review in order to perform the test against all annotation types.

### TODO

- [x] common test case for enter event for all annotation types
- [x] common test case for click event for all annotation types
- [x] common test case for dblclick event for all annotation types
- [x] common test case for leave event for all annotation types

### CODE REVIEW

**Comment by kurkle** [comment]
Looks like a good idea 👍

**Comment by stockiNail** [comment]
> Looks like a good idea 👍

Thank you!! I don't know if you have seen but from version 1.1.0 to the current master, the coverage has been increased from 9% to 86%!! Very good (I guess)!

**Comment by kurkle** [comment]
> Thank you!! I don't know if you have seen but from version 1.1.0 to the current master, the coverage has been increased from 79% to 86%!! Very good (I guess)!

Yes, very good!

**Comment by stockiNail** [comment]
It's not clear to me how coveralls is checking the tests.

Example: Ellipse

image

It seems that `inRange` is never tested but that's not true because is invoked for events and there is a special test case to test it directly....

It's weird.

**Comment by kurkle** [comment]
> It's not clear to me how coveralls is checking the tests.
> 
> Example: Ellipse
> 
> image
> 
> It seems that `inRange` is never tested but that's not true because is invoked for events and there is a special test case to test it directly....
> 
> It's weird.

I'll have to look at this closer, to see why that happens.

**Comment by stockiNail** [comment]
> I'll have to look at this closer, to see why that happens.

II think you shouldn't take time for that. I was just curious about this weird behavior. Anyway the result is quite good, because now the master reached 87%, this PR will add another 4.9% therefore I expect to exceed 91/92% (+12/13% from version 1.1).

File: test/specs/ellipse.spec.js [hunk 1]
```diff
@@ -1,3 +1,16 @@
 describe('Ellipse annotation', function() {
   describe('auto', jasmine.fixtures('ellipse'));
+
+  window.testEvents({
+    type: 'ellispe',
```
Comment (kurkle) [inline test/specs/ellipse.spec.js:5]: Took a closer look anyway and here is the problem with ellipse (its not ellispe) :)

Comment (kurkle) [reply test/specs/ellipse.spec.js:5]: This leads me to think that we should at least output a warning when the type is not found.

Comment (stockiNail) [reply test/specs/ellipse.spec.js:5]: azz!!!! THANK YOU!

Comment (stockiNail) [reply test/specs/ellipse.spec.js]: fixed

Comment (stockiNail) [reply test/specs/ellipse.spec.js]: > This leads me to think that we should at least output a warning when the type is not found.

let me add it

**Review by kurkle — APPROVED** [review]

File: test/specs/ellipse.spec.js (hunk 1 above)
Comment (kurkle) [reply test/specs/ellipse.spec.js]: Lets do that separately

**Pull request merged**